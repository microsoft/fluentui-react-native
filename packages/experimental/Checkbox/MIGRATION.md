# Checkbox Migration

## STATUS: WIP ðŸš§

This Migration guide is a work in progress and is not yet ready for use.

## Migration from v0 Checkbox

### Component renames

| v0 `Checkbox`  | v1 `Checkbox`                       |
| -------------- | ----------------------------------- |
| `<Checkbox />` | `<Checkbox>Hello, world</Checkbox>` |

### Props that remain as is

- Any props that are part of `ViewProps`
- `children`
- `componentRef`
- `icon`
- `onClick`
- `testID`
- `tooltip`

### Props no longer supported with an equivalent functionality in v1 Checkbox

- `content` => Pass the content as `children` instead
- `endIcon` => Use `after` value for `iconPosition` prop and pass icon information into `icon` prop instead
- `startIcon` => Use `before` value for `iconPosition` prop and pass icon information into `icon` prop instead

### Props no longer supported without an equivalent functionality in v1 Checkbox

- Cannot use both `startIcon` and `endIcon` at the same time in v2

### Tokens that remain as is

- Any props that are part of `FontTokens`, `IForegroundColorTokens`, `IBackgroundColorTokens`, `IBorderTokens`
- `iconColor`
- `iconSize`
- `iconWeight`

### Tokens no longer supported with an equivalent functionality in v1 Checkbox

- `contentPadding` => Set on `style` property of `root` property under `slotProps` in `compose` API
- `iconColorHovered` => Use `iconColor` token under the `hovered` state token
- `iconColorPressed` => Use `iconColor` token under the `pressed` state token

### Tokens no longer supported without an equivalent functionality in v1 Checkbox

- `contentPaddingFocused` => Removed in favor of using native based focus visuals
- `wrapperBorderColor` => Removed in favor of using native based focus visuals

### Slots no longer supported with an equivalent functionality in v1 Checkbox

- `borderWrapper` has been removed in favor of using native focus visuals. Use `root` instead.
- `stack` has been removed. Use `root` instead.

### Updating ThemeProvider

If you are using the older theme provider `ThemeProvider` from `@uifabricshared/theming-react-native`, you will need to update the `ThemeProvider` to pull from `@fluentui-react-native/theme` to have the control work properly with themes. Please see [this page](../../../docs/pages/Guides/UpdateThemeProvider.md) for guidance.

### Migrating customized Checkboxes

Please see [this page](../../../docs/pages/Guides/UpdatingCustomize.md) for guidance on how to move from the old `customize` API to the new one.

### Migrating composed Checkboxes

Please see [this page](../../../docs/pages/Guides/UpdatingCustomize.md) for guidance on how to move from the old `customize` API to the new one.

## Porting from FluentUI v9 Checkbox

The FURN Checkbox cannot be used in place of a FluentUI Checkbox - these buttons are intended to be used on their respective platforms. See [this porting guide](../../../docs/pages/Guides/PortingFromFluentUI.md) for general guidance on coming from FluentUI to FURN.

### Props that remain as is

- `children`
- `disabled`
- `icon`
- `onClick`
- `testID`
- `tooltip`

### Props renamed to align with ReactNative

No `Checkbox` specific renames.

### Prop differences due to technical differences and limitations

- `icon` takes in a props object instead of the JSX element itself. This is due to framework differences from FluentUI.
- `iconOnly` must be supplied for components do not have any text content for them to be styled correctly. This is due to framework differences from FluentUI.
- `ref` is exposed as `componentRef`, similar to previous versions of FluentUI. This is due to framework differences from FluentUI.

### Other Prop differences

- `appearance` does not include values for `transparent` or `outline` as neither currently are used by native platforms.
- `iconPosition` uses `before` and `after` as values instead of `left` and `right` for better right to left support.

## Property mapping

| v0 `Checkbox`  | v1 `Checkbox`  |
| -------------- | -------------- |
| `componentRef` | `componentRef` |
| `content`      |                |
| `disabled`     | `disabled`     |
| `endIcon`      | `iconPosition` |
| `icon`         | `icon`         |
| `onClick`      | `onClick`      |
| `statIcon`     | `iconPosition` |
| `testId`       | `testId`       |
| `tooltip`      | `tooltip`      |
