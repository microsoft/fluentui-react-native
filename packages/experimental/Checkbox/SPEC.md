# Checkbox

## Background

The `Checkbox` component enables users to select one or more items from a group, or switch between two mutually exclusive options (checked or unchecked).

## Requirements

If using FURN's theming, the `Checkbox` requires use of the `ThemeProvider` from `@fluentui-react-native/theme` to work properly with themes. Please see [this page](https://github.com/microsoft/fluentui-react-native/blob/master/docs/pages/Guides/UpdateThemeProvider.md) for information on updating your `ThemeProvider` if using the version from `@uifabricshared/theming-react-native`.

## Sample Code

Basic examples:

```jsx
<Checkbox label="Example Checkbox" />
<Checkbox label="Circular Checkbox" circular size="large" />
<Checkbox label="Controlled Checkbox" onChange={onChangeFunction} checked={checked} />
```

More examples on the [Test pages for the Checkbox](https://github.com/microsoft/fluentui-react-native/tree/master/apps/fluent-tester/src/FluentTester/TestComponents/CheckboxExperimental). Instructions on running the tester app can be found [here](https://github.com/microsoft/fluentui-react-native/blob/master/apps/fluent-tester/README.md).

## Visual Examples

Win32:

![Checkbox with text on win32 example](./assets/Checkbox_example_win32.png)

```tsx
<Checkbox>Text</Checkbox>
```

![Checkbox with text and primary appearance on win32 example](./assets/Checkbox_primary_example_win32.png)

```tsx
<Checkbox appearance="primary">Text</Checkbox>
```

## Variants

### Appearance

The `Checkbox` component has several apparance variants depending on where it's being used:

- The default `Checkbox` is rendered with its default styling indicating a trigger for an action.
- appearance="primary": The `Checkbox` is styled to emphasize that it represents the primary action.
- appearance="subtle": The `Checkbox` is styled to blend into its background to become less emphasized.

### Icon

The `Checkbox` component can include an `icon` that appears before or after its `children`. If an `icon` is provided without any other `children` passed into `Checkbox`, then the `Checkbox` becomes an icon-only `Checkbox`.

### Shape

- shape="rounded": The Checkbox as rounded corners. This is the default if shape is not set.
- shape="circular": The Checkbox has completely round corners. A Checkbox of equal width and height will be a circle.
- shape="square": The Checkbox has right-angle corners.

### Sizes

The `Checkbox` component supports different sizing with at least three different sizes: `small`, `medium`, and `large`. `Small` is the default on `win32`, `medium` is the default on other platforms.

### Block

The `Checkbox` component can completely fill the width of its container.

### Loading

The `Checkbox` component can be loading if it's waiting for another action to occur before allowing itself to be interacted with.

## API

### Slots

The `Checkbox` component has three slots, or parts. The slots behave as follows:

- `root` - The outer container representing the `Checkbox` itself that wraps everything passed via the `children` prop.
- `icon` - If specified, renders an `icon` either before or after the `children` as specified by the `iconPosition` prop.
- `content` - If specified, renders the first entry of `children` as text.

The slots can be modified using the `compose` function on the `Checkbox`. For more information on using the `compose` API, please see [this page](../../framework/composition/README.md).

### Props

Below is the set of props the Checkbox supports:

```ts
export interface CheckboxProps extends Omit<IWithPressableOptions<ViewProps>, 'onPress'> {
  /**
   * A Checkbox can have its content and borders styled for greater emphasis or to be subtle.
   * - 'primary': Emphasizes the Checkbox as a primary action.
   * - 'subtle': Minimzes emphasis to blend into the background until hovered or focused.
   */
  appearance?: 'primary' | 'subtle';

  /**
   * A Checkbox can fill the width of its container.
   * @default false
   */
  block?: boolean;

  /**
   * A RefObject to access the ICheckbox interface. Use this to access the public methods and properties of the component.
   */
  componentRef?: React.RefObject<IFocusable>;

  /**
   * Icon slot that, if specified, renders an icon either before or after the `children` as specified by the
   * `iconPosition` prop.
   */
  icon?: IconSourcesType;

  /**
   * Checkbox contains only icon, there's no text content
   * Must be set for Checkbox to style correctly when Checkbox has not content.
   */
  iconOnly?: boolean;

  /**
   * A Checkbox can format its icon to appear before or after its content.
   * @default 'before'
   */
  iconPosition?: 'before' | 'after';

  /**
   * A Checkbox can show a loading indicator if it is waiting for another action to happen before allowing itself to
   * be interacted with.
   * @default false
   */
  loading?: boolean;

  /**
   * A Checkbox can be rounded, circular, or square.
   * @default 'rounded'
   */
  shape?: 'rounded' | 'circular' | 'square';

  /**
   * A Checkbox supports different sizes.
   * @default 'medium'
   */
  size?: 'small' | 'medium' | 'large';

  /**
   * Text that should show in a tooltip when the user hovers over a Checkbox.
   */
  tooltip?: string;

  /**
   * A callback to call on Checkbox click event
   */
  onClick?: () => void;
}
```

### Styling Tokens

Tokens can be used to customize the styling of the control by using the `customize` function on the `Checkbox`. For more information on using the `customize` API, please see [this page](../../framework/composition/README.md). The `Checkbox` has the following tokens:

```ts
export interface CheckboxTokens extends LayoutTokens, FontTokens, IBorderTokens, IShadowTokens, IColorTokens {
  /**
   * The icon color.
   */
  iconColor?: ColorValue;

  /**
   * The size of the icon.
   */
  iconSize?: number;

  /**
   * The weight of the lines used when drawing the icon.
   */
  iconWeight?: number;

  /**
   * The width of the Checkbox.
   */
  width?: ViewStyle['width'];

  /**
   * The amount of spacing between an icon and the content when iconPosition is set to 'before', in pixels
   */
  spacingIconContentBefore?: number;

  /**
   * The amount of spacing between an icon and the content when iconPosition is set to 'after', in pixels
   */
  spacingIconContentAfter?: number;

  /**
   * States that can be applied to a Checkbox.
   * These can be used to modify styles of the Checkbox when under the specified state.
   */
  hovered?: CheckboxTokens;
  focused?: CheckboxTokens;
  pressed?: CheckboxTokens;
  disabled?: CheckboxTokens;
  hasContent?: CheckboxTokens;
  hasIconAfter?: CheckboxTokens;
  hasIconBefore?: CheckboxTokens;
  primary?: CheckboxTokens;
  subtle?: CheckboxTokens;
  block?: CheckboxTokens;
  small?: CheckboxTokens;
  medium?: CheckboxTokens;
  large?: CheckboxTokens;
  rounded?: CheckboxTokens;
  circular?: CheckboxTokens;
  square?: CheckboxTokens;
}
```

## Behaviors

### States

The following section describes the different states which `Checkbox` can be in as a result of interaction.

#### Enabled and Disabled states

An enabled `Checkbox` communicates interaction by having styling that invites the user to click/tap on it to trigger an action.

A disabled `Checkbox` is non-interactive, disallowing the user to click/tap on it to trigger an action.

#### Hovered state

A hovered `Checkbox` changes styling to communicate that the user has placed a cursor above it.

#### Focused state

A focused `Checkbox` changes styling to communicate that the user has placed keyboard focus on it. This styling is usually the same to the one in the hovered state plus extra styling on the outline to indicate keyboard focus has been placed on the component.

#### Pressed state

A pressed `Checkbox` changes styling to communicate that the user is currently pressing it.

#### Loading state

A loading `Checkbox` renders a `loader` before all the other content to indicate that it is waiting for another action before allowing itself to be interacted with.

### Interaction

#### Keyboard interaction

The following is a set of keys that interact with the `Checkbox` component:

| Key     | Description                                           |
| ------- | ----------------------------------------------------- |
| `Enter` | Executes the function passed into the `onClick` prop. |
| `Space` | Executes the function passed into the `onClick` prop. |

#### Cursor interaction

- Cursor moves onto botton: Should immediately change the styling of the `Checkbox` so that it appears to be hovered.
- Cursor moves out of botton: Should immediately remove the hovered styling of the `Checkbox`.
- Mouse click: Should execute the `Checkbox` and move focus to its target.

#### Touch interaction

The same behavior as above translated for touch events. This means that there is no equivalent for `onHoverIn` and `onHoverOut`, which makes it so that the hovered state cannot be accessed.

## Accessibility

### Expected behavior

- Should default to adding `role="Checkbox"` to the root slot.
- Should mix in the accessibility props expected for a `Checkbox` component.
- Should be keyboard tabbable and focusable.

See [`useCheckbox` hook](https://github.com/microsoft/fluentui-react-native/blob/master/packages/experimental/Checkbox/src/useCheckbox.ts) for details on accessibility props
