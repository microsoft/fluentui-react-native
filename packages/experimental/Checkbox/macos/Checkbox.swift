class Checkbox: NSButton {

	@objc public var onPress: RCTBubblingEventBlock?

	public override init(frame:NSRect) {
		super.init(frame: frame)
	}

	required init?(coder: NSCoder) {
		preconditionFailure("init(coder:) has not been implemented")
	}

	@objc public func sendCallback(sender: NSButton) {
		if (onPress != nil) {
			let isChecked = self.state == NSControl.StateValue.on
			onPress!(["isChecked": isChecked]);
		}
	}
}
