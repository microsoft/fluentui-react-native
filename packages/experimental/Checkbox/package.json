{
  "name": "@fluentui-react-native/experimental-checkbox",
  "version": "0.4.7",
  "description": "A cross-platform Checkbox component using the Fluent Design System",
  "main": "src/index.ts",
  "module": "src/index.ts",
  "typings": "lib/index.d.ts",
  "onPublish": {
    "main": "lib-commonjs/index.js",
    "module": "lib/index.js"
  },
  "scripts": {
    "build": "fluentui-scripts build",
    "clean": "fluentui-scripts clean",
    "depcheck": "fluentui-scripts depcheck",
    "just": "fluentui-scripts",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "prettier": "fluentui-scripts prettier",
    "prettier-fix": "fluentui-scripts prettier --fix true"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui-react-native.git",
    "directory": "packages/experimental/Checkbox"
  },
  "dependencies": {
    "@fluentui-react-native/adapters": ">=0.8.2 <1.0.0",
    "@fluentui-react-native/experimental-text": ">=0.7.6 <1.0.0",
    "@fluentui-react-native/component-cache": "^1.3.1",
    "@fluentui-react-native/framework": "0.7.6",
    "@fluentui-react-native/interactive-hooks": ">=0.11.7 <1.0.0",
    "@fluentui-react-native/pressable": ">=0.8.7 <1.0.0",
    "@fluentui-react-native/theme-tokens": "^0.11.2",
    "@fluentui-react-native/tokens": ">=0.11.3 <1.0.0",
    "@fluentui-react-native/use-styling": ">=0.8.2 <1.0.0",
    "react-native-svg": "^12.1.1",
    "tslib": "^2.3.1"
  },
  "devDependencies": {
    "@fluentui-react-native/eslint-config-rules": "^0.1.1",
    "@office-iss/react-native-win32": "^0.64.8",
    "@types/react-native": "^0.64.0",
    "@uifabricshared/build-native": "^0.1.1",
    "react": "17.0.1",
    "react-native": "^0.64.3"
  },
  "peerDependencies": {
    "react": ">=17.0.1",
    "react-native": ">=0.64.3"
  },
  "author": "",
  "license": "MIT"
}
