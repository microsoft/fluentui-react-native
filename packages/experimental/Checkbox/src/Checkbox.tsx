/** @jsxRuntime classic */
/** @jsx withSlots */
import { CheckboxV1 } from '@fluentui-react-native/checkbox';

if (__DEV__) {
  console.warn(
    'The @fluentui-react-native/exprimental-checkbox package is deprecated for win32. The contents of this package have been moved to @fluentui-react-native/checkbox. If you need to use the Checkbox component from this package, please use CheckboxV1 from @fluentui-react-native/checkbox.',
  );
}

export const Checkbox = CheckboxV1;

export default Checkbox;
