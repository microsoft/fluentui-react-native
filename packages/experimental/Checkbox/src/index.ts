export type {
  CheckboxInfo,
  CheckboxProps,
  CheckboxSize,
  CheckboxShape,
  CheckboxSlotProps,
  CheckboxState,
  CheckboxTokens,
  CheckboxType,
} from '@fluentui-react-native/checkbox';
export { checkboxNameV1 as checkboxName } from '@fluentui-react-native/checkbox';
export { Checkbox } from './Checkbox';
