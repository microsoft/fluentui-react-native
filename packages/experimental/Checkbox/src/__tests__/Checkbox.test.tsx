import * as React from 'react';
import { Checkbox } from '../Checkbox';
import * as renderer from 'react-test-renderer';

function onChange(isChecked: boolean) {
  console.log(isChecked);
}

describe('Checkbox component tests', () => {
  it('Checkbox default', () => {
    const tree = renderer.create(<Checkbox label="Default Checkbox" onChange={onChange} defaultChecked={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox all props', () => {
    const tree = renderer
      .create(
        <Checkbox
          label="All Props Checkbox"
          onChange={onChange}
          defaultChecked={true}
          labelPosition="before"
          disabled
          circular
          size="large"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checkbox all tokens', () => {
    const BoldCheckbox = Checkbox.customize({
      fontFamily: 'Wingdings',
      fontWeight: '900',
      fontSize: 20,
      checkboxBackgroundColor: 'blue',
      checkboxBorderColor: 'red',
      checkboxBorderRadius: 5,
      checkboxBorderWidth: 2,
      checkmarkColor: 'pink',
      checkmarkOpacity: 0.7,
      backgroundColor: 'purple',
      color: 'yellow',
      spacingLabelAfter: 7,
      borderColor: 'green',
      borderRadius: 4,
      borderWidth: 1,
    });
    const tree = renderer.create(<BoldCheckbox label="All Tokens Checkbox" onChange={onChange} defaultChecked={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
