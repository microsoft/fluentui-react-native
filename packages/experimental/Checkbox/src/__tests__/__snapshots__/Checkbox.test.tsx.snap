// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component tests Checkbox all props 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="All Props Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": true,
      "required": true,
    }
  }
  accessible={true}
  defaultChecked={true}
  disabled={true}
  enableFocusRing={true}
  focusable={false}
  labelPosition="before"
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  shape="circular"
  size="large"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#a19f9d",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "margin": 0,
        "marginEnd": 12,
      }
    }
  >
    All Props Checkbox
  </Text>
  <Text
    style={
      Object {
        "color": "#690a19",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "margin": 0,
        "paddingStart": 8,
      }
    }
  >
    *
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#f3f2f1",
        "borderRadius": 9999,
        "borderWidth": 1,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={10}
      bbWidth={10}
      color={-4342339}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#bdbdbd",
            "height": 10,
            "opacity": 1,
            "width": 10,
          },
          Object {
            "flex": 0,
            "height": 10,
            "width": 10,
          },
        ]
      }
      tintColor={-4342339}
      vbHeight={8}
      vbWidth={11}
    >
      <RNSVGGroup
        opacity={1}
      >
        <RNSVGPath
          d="M10.2637 1.26367L4 7.5332L0.736328 4.26367L1.26367 3.73633L4 6.4668L9.73633 0.736328L10.2637 1.26367Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Checkbox component tests Checkbox all tokens 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="All Tokens Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  defaultChecked={false}
  enableFocusRing={true}
  focusable={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "purple",
      "borderColor": "green",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderColor": "red",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={8}
      bbWidth={8}
      color={-16181}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "pink",
            "height": 8,
            "opacity": 0.7,
            "width": 8,
          },
          Object {
            "flex": 0,
            "height": 8,
            "width": 8,
          },
        ]
      }
      tintColor={-16181}
      vbHeight={8}
      vbWidth={11}
    >
      <RNSVGGroup
        opacity={0.7}
      >
        <RNSVGPath
          d="M10.2637 1.26367L4 7.5332L0.736328 4.26367L1.26367 3.73633L4 6.4668L9.73633 0.736328L10.2637 1.26367Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "yellow",
        "fontFamily": "Wingdings",
        "fontSize": 20,
        "fontWeight": "900",
        "margin": 0,
        "marginEnd": 9,
        "marginStart": 12,
      }
    }
  >
    All Tokens Checkbox
  </Text>
</View>
`;

exports[`Checkbox component tests Checkbox composed 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Composed Button with RNText"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": undefined,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#323130",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={8}
      bbWidth={8}
      color={-14672098}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#201f1e",
            "height": 8,
            "opacity": 0,
            "width": 8,
          },
          Object {
            "flex": 0,
            "height": 8,
            "width": 8,
          },
        ]
      }
      tintColor={-14672098}
      vbHeight={8}
      vbWidth={11}
    >
      <RNSVGGroup
        opacity={0}
      >
        <RNSVGPath
          d="M10.2637 1.26367L4 7.5332L0.736328 4.26367L1.26367 3.73633L4 6.4668L9.73633 0.736328L10.2637 1.26367Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "marginStart": 12,
      }
    }
  >
    Composed Button with RNText
  </Text>
</View>
`;

exports[`Checkbox component tests Checkbox default 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Default Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  defaultChecked={false}
  enableFocusRing={true}
  focusable={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "display": "flex",
      "flexDirection": "row",
      "padding": 8,
      "paddingHorizontal": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#323130",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={8}
      bbWidth={8}
      color={-14672098}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#201f1e",
            "height": 8,
            "opacity": 0,
            "width": 8,
          },
          Object {
            "flex": 0,
            "height": 8,
            "width": 8,
          },
        ]
      }
      tintColor={-14672098}
      vbHeight={8}
      vbWidth={11}
    >
      <RNSVGGroup
        opacity={0}
      >
        <RNSVGPath
          d="M10.2637 1.26367L4 7.5332L0.736328 4.26367L1.26367 3.73633L4 6.4668L9.73633 0.736328L10.2637 1.26367Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "margin": 0,
        "marginStart": 12,
      }
    }
  >
    Default Checkbox
  </Text>
</View>
`;

exports[`Checkbox component tests Checkbox no label 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
      "required": false,
    }
  }
  accessible={true}
  defaultChecked={false}
  enableFocusRing={true}
  focusable={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": undefined,
      "borderRadius": 2,
      "display": "flex",
      "flexDirection": "row",
      "padding": 4,
      "paddingHorizontal": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#323130",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={8}
      bbWidth={8}
      color={-14672098}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#201f1e",
            "height": 8,
            "opacity": 0,
            "width": 8,
          },
          Object {
            "flex": 0,
            "height": 8,
            "width": 8,
          },
        ]
      }
      tintColor={-14672098}
      vbHeight={8}
      vbWidth={11}
    >
      <RNSVGGroup
        opacity={0}
      >
        <RNSVGPath
          d="M10.2637 1.26367L4 7.5332L0.736328 4.26367L1.26367 3.73633L4 6.4668L9.73633 0.736328L10.2637 1.26367Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "400",
        "margin": 0,
      }
    }
  />
</View>
`;
